web/auth/auth.go

```diff
@@ -49,20 +49,27 @@ func newUserSession(id uint64) *UserSession {
 
 // UserToken holds a map from session cookies to user IDs.
 type UserToken struct {
-	store map[string]uint64
+	store map[string]*http.Request
 }
 
-func (ut *UserToken) add(token string, id uint64) {
-	for token, userID := range TokenStore.store {
-		if userID == id {
-			delete(TokenStore.store, token)
+func (ut *UserToken) add(token string) {
+	if ut.store[token] == nil {
+		if request, err := http.NewRequest("GET", "/", nil); err == nil {
+			request.Header.Set(Cookie, token)
+			ut.store[token] = request
 		}
 	}
-	ut.store[token] = id
 }
 
 func (ut *UserToken) Get(token string) uint64 {
-	return TokenStore.store[token]
+	ut.add(token)
+	if sess, err := gothic.Store.Get(ut.store[token], SessionKey); err == nil {
+		if i, ok := sess.Values[UserKey]; ok {
+			us := i.(*UserSession)
+			return us.ID
+		}
+	}
+	return 0
 }
 
 var TokenStore *UserToken
```

Comment:
Suggest to move this line between the `UserToken` struct declaration and the `add` method.